sonaranalyzer-dotnet/src/SonarAnalyzer.CSharp/Rules/UseUriInsteadOfString.cs

```diff
@@ -106,6 +112,33 @@ protected sealed override void Initialize(SonarAnalysisContext context)
                     }
                 },
                 SyntaxKind.PropertyDeclaration);
+
+            context.RegisterSyntaxNodeActionInNonGenerated(
+                c =>
+                {
+                    var invokedMethod = c.SemanticModel.GetSymbolInfo(c.Node).Symbol as IMethodSymbol;
+                    if (invokedMethod == null)
+                    {
+                        return;
+                    }
+
+                    if (invokedMethod.IsInType(KnownType.System_Uri))
```

Comment:
This is to prevent Uri methods - things like Uri.TryCreate() from being flagged. What's your opinion on excluding this from checks? Should we update RSPEC?